tower-http/src/redirect.rs

```diff
@@ -0,0 +1,100 @@
+//! Service that redirects all requests.
+
+use http::{header, HeaderValue, Response, StatusCode, Uri};
+use std::{
+    convert::Infallible,
+    future::Future,
+    marker::PhantomData,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_service::Service;
+
+/// Service that redirects all requests.
+#[derive(Clone, Debug)]
+pub struct Redirect<ResBody> {
+    status_code: StatusCode,
+    uri: Uri,
+    _marker: PhantomData<fn() -> ResBody>,
+}
+
+impl<ResBody> Redirect<ResBody> {
+    /// Create a new [`Redirect`] that uses a `307 Temporary Redirect` status code.
+    pub fn temporary(uri: Uri) -> Self {
+        Self::with_status_code(StatusCode::TEMPORARY_REDIRECT, uri)
+    }
+
+    /// Create a new [`Redirect`] that uses a `308 Permanent Redirect` status code.
```

Comment:
nit, take it or leave it: might be nice to link to docs for these status codes here --- maybe MDN?

Reply:
👍